.circleci/config.yml

```diff
@@ -20,36 +20,31 @@ jobs:
     steps:
       - checkout
       - run:
-          name: Run pre-compile checks
+          name: Run pre-build checks
           command: ./scripts/run-pre-build-checks
       - restore_cache:
-          # Adding `-v4` suffix to enable easy cache invalidation when needed (just change to `-v<next-integer>`)
+          # Adding `-v<N>` suffix to enable easy cache invalidation when needed:
+          # just change to `-v<N+1>` (remember to also update `save_cache` key).
           keys:
             # First, try finding a cache entry with the same set of libraries
             # that also comes from the same branch (so as to make the best use of Gradle compilation cache).
-            - gradle-deps-v4-{{ checksum "build.gradle" }}-{{ .Branch }}
+            - gradle-deps-v5-{{ checksum "gradle/libs.versions.toml" }}-{{ .Branch }}
```

Comment:
Cache key computed out of library versions that actually matter for the sake of caching